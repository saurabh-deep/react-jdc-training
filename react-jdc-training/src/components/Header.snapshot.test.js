import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Header from "./Header";

describe("Header", () => {
  const headerText = "Testing header component text";
  const clickHandler = jest.fn();

  it("should render properly", () => {
    const wrapper = shallow(
      <Header title={headerText} />
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("should handle a click event", () => {
    const wrapper = shallow(
      <Header title={headerText} onHeaderClick={clickHandler} />
    );
    const anchor = wrapper.find("#logo");
    anchor.simulate("click");
    expect(clickHandler).toHaveBeenCalled();
  });
});