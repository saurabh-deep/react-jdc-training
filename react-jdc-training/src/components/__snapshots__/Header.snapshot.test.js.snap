// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<div
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <a
    href="/"
    id="logo"
  >
    <h2>
      Testing header component text
    </h2>
  </a>
</div>
`;
